pygls/server.py

```diff
@@ -407,9 +407,17 @@ def progress(self) -> Progress:
         """Gets the object to manage client's progress bar."""
         return self.lsp.progress
 
-    def publish_diagnostics(self, doc_uri: str, diagnostics: List[Diagnostic]):
-        """Sends diagnostic notification to the client."""
-        self.lsp.publish_diagnostics(doc_uri, diagnostics)
+    def publish_diagnostics(
+        self,
+        uri: str,
+        diagnostics: Optional[List[Diagnostic]] = None,
+        version: Optional[int] = None,
+        **kwargs
+    ):
+        """
+        Sends diagnostic notification to the client.
+        """
+        self.lsp.publish_diagnostics(uri, diagnostics, version, **kwargs)
```

Comment:
Just a thought.

It might make sense to have this method construct and pass a `PublishDiagnosticsParams` object, so that the user doesn't see a deprecation warning on what appears to be an internal function.

But perhaps that can wait until we decide on #306

Reply:
This is a really good point. I've just force-pushed this. Is it what you meant?

(Apologies for the delay too! I'm in South America at the moment so it was hooooot holiday season 🥵 hahah)